src/LayerEditor/ui/diagram_editor/diagram_scene.py

```diff
@@ -112,7 +110,7 @@ def add_point(self, pos, gitem):
             if pt.id in self.points:
                 gpt = self.points[pt.id]
             else:
-                gpt = GsPoint(pt)
+                gpt = GsPoint(pt, self.block)
```

Comment:
Is the `self.block` really necessary there.

Reply:
Similar for GsSegment and GsPolygon.

Reply:
Yes it is, because it is used in update to get currently active layer. Update is overridden QT function so it cannot be passed as parameter.